modules/redis/src/main/scala/zio/redis/internal/SubscribeEnvironment.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 John A. De Goes and the ZIO contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.redis.internal
+
+import zio.redis.{CodecSupplier, SubscriptionExecutor}
+import zio.schema.Schema
+import zio.schema.codec.BinaryCodec
+
+private[redis] trait SubscribeEnvironment {
```

Comment:
We might be able to generalize the environments, though I'm fine with exploring that possibility in a follow-up.